clients/src/main/java/org/apache/kafka/common/requests/FetchRequest.java

```diff
@@ -277,14 +277,18 @@ public AbstractResponse getErrorResponse(int throttleTimeMs, Throwable e) {
         // is essential for them.
         Errors error = Errors.forException(e);
         List<FetchResponseData.FetchableTopicResponse> topicResponseList = new ArrayList<>();
-        data.topics().forEach(topic -> {
-            List<FetchResponseData.PartitionData> partitionResponses = topic.partitions().stream().map(partition ->
-                    FetchResponse.partitionResponse(partition.partition(), error)).collect(Collectors.toList());
-            topicResponseList.add(new FetchResponseData.FetchableTopicResponse()
-                    .setTopic(topic.topic())
-                    .setTopicId(topic.topicId())
-                    .setPartitions(partitionResponses));
-        });
+        // Since UNKNOWN_TOPIC_ID is a new error type only returned when topic ID requests are made (from newer clients),
```

Comment:
We need to do something like this to easily get the top level error with no partition response for UNKNOWN_TOPIC_ID. I think this works, but we may want a version check as well just to be safe.